app/services/filter_tagged_content.rb

```diff
@@ -0,0 +1,62 @@
+class FilterTaggedContent
+  def valid?(document)
+    return valid_document_collection?(document) if document.document_collection?
+
+    valid_content_item?(document)
+  end
+
+private
+
+  def valid_document_collection?(document)
+    return true unless all_document_collections.include?(document.base_path)
+
+    valid_document_collections.include?(document.base_path)
+  end
+
+  def valid_content_item?(document)
+    return true unless document.tagged_to_document_collection?
+
+    document.document_collections.all? do |document_collection|
+      valid_content_items_in_document_collections.include?(
+        document_collection['link']
+      )
+    end
+  end
+
+  def valid_document_collections
```

Comment:
It's only being used in one place, I'm tempted to leave it as is for now if that's ok.